### ISSUE TITLE
Packer amazon-ebs builder cannot ssh with Amazon Linux 2022

### ISSUE THREAD

**Issue by amrragab8080** [hashicorp/packer-plugin-amazon#172]
Packer cannot ssh handshake with AL2022 
```
"builders": [{
    "type": "amazon-ebs",
    "region": "{{user `region`}}",
    "source_ami": "ami-001e76b3918fba080",
    "run_tags": {
        "Name": "packer-gpu-processor_{{user `flag`}}"
    },
    "subnet_id": "{{user `subnet_id`}}",
    "security_group_ids": "{{user `security_groupids`}}",
    "instance_type":"r5n.12xlarge",
    "ssh_username": "ec2-user",
```
PACKER_LOG
I am using packer's temporary keypair
==> amazon-ebs: Creating temporary keypair: packer_61d7e58d-3b50-fac8-81af-f5de9430eb01
```
2022/01/07 01:57:42 packer-builder-amazon-ebs plugin: [DEBUG] Detected authentication error. Increasing handshake attempts.
2022/01/07 01:57:49 packer-builder-amazon-ebs plugin: Using host value: 3.91.73.151
2022/01/07 01:57:49 packer-builder-amazon-ebs plugin: [INFO] Attempting SSH connection to 3.91.73.151:22...
2022/01/07 01:57:49 packer-builder-amazon-ebs plugin: [DEBUG] reconnecting to TCP connection for SSH
2022/01/07 01:57:49 packer-builder-amazon-ebs plugin: [DEBUG] handshaking with SSH
2022/01/07 01:57:49 packer-builder-amazon-ebs plugin: [DEBUG] SSH handshake err: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
```

If i switch to a AL2 AMI works fine.

Packer 1.7.8

**Comment by wedge-jarrad** [comment]
I ran into the same thing. Creating an al2022 ec2 manually and trying to SSH from a Windows box resulted in the same thing. I did notice that it did work manually when using an ed25519 keypiar, though.

So a workaround would be to set `temporary_key_pair_type = "ed25519"`. Per #144 this doesn't work just yet, but I just submitted a PR which will hopefully close that out soon. Here's output from my test script showing it in action. Note that it's just a shell provisioner which runs `date` and `uname -a`, the latter showing the amzn2022 kernel.

```
whatever.amazon-ebs.al2022_x86_64: output will be in this color.

==> whatever.amazon-ebs.al2022_x86_64: Prevalidating any provided VPC information
==> whatever.amazon-ebs.al2022_x86_64: Prevalidating AMI Name: jarrad-144-test
    whatever.amazon-ebs.al2022_x86_64: Found Image ID: ami-044065b5480679567
==> whatever.amazon-ebs.al2022_x86_64: Creating temporary keypair: packer_61e8fbf4-90fa-9882-72a3-2b85a48450c0
==> whatever.amazon-ebs.al2022_x86_64: Creating temporary security group for this instance: packer_61e8fbf5-e78b-2ccf-b8f2-25b5ac3a1e21
==> whatever.amazon-ebs.al2022_x86_64: Authorizing access to port 22 from [0.0.0.0/0] in the temporary security groups...
==> whatever.amazon-ebs.al2022_x86_64: Launching a source AWS instance...
==> whatever.amazon-ebs.al2022_x86_64: Adding tags to source instance
    whatever.amazon-ebs.al2022_x86_64: Adding tag: "Name": "Packer Builder"
    whatever.amazon-ebs.al2022_x86_64: Instance ID: i-0315f0cf4f603e8bd
==> whatever.amazon-ebs.al2022_x86_64: Waiting for instance (i-0315f0cf4f603e8bd) to become ready...
==> whatever.amazon-ebs.al2022_x86_64: Using SSH communicator to connect: 35.84.178.131
==> whatever.amazon-ebs.al2022_x86_64: Waiting for SSH to become available...
==> whatever.amazon-ebs.al2022_x86_64: Connected to SSH!
==> whatever.amazon-ebs.al2022_x86_64: Provisioning with shell script: /tmp/packer-shell946114196
    whatever.amazon-ebs.al2022_x86_64: Thu Jan 20 06:07:25 AM UTC 2022
    whatever.amazon-ebs.al2022_x86_64: Linux ip-172-31-53-14.us-west-2.compute.internal 5.10.75-82.359.amzn2022.x86_64 #1 SMP Mon Nov 15 18:04:59 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
==> whatever.amazon-ebs.al2022_x86_64: Stopping the source instance...
    whatever.amazon-ebs.al2022_x86_64: Stopping instance
==> whatever.amazon-ebs.al2022_x86_64: Waiting for the instance to stop...
==> whatever.amazon-ebs.al2022_x86_64: Skipping AMI creation...
==> whatever.amazon-ebs.al2022_x86_64: Skipping Enable AMI deprecation...
==> whatever.amazon-ebs.al2022_x86_64: Skipping AMI region copy...
==> whatever.amazon-ebs.al2022_x86_64: Skipping AMI modify attributes...
==> whatever.amazon-ebs.al2022_x86_64: Skipping AMI create tags...
==> whatever.amazon-ebs.al2022_x86_64: Terminating the source AWS instance...
==> whatever.amazon-ebs.al2022_x86_64: Cleaning up any extra volumes...
==> whatever.amazon-ebs.al2022_x86_64: No volumes to clean up, skipping
==> whatever.amazon-ebs.al2022_x86_64: Deleting temporary security group...
==> whatever.amazon-ebs.al2022_x86_64: Deleting temporary keypair...
Build 'whatever.amazon-ebs.al2022_x86_64' finished after 1 minute 14 seconds.

==> Wait completed after 1 minute 14 seconds

==> Builds finished but no artifacts were created.
```

In the short term manually providing an ed25519 keypair with `ssh_private_key_file` may also work, though I haven't tested that yet.

**Comment by Xerkus** [comment]
Amazon Linux is now based on Fedora Linux afaik. Setting crypto policy to legacy could be used as a workaround:

```
user_data = <<EOF
#!/bin/bash
update-crypto-policies --set LEGACY
EOF
```

**Comment by programmablemike** [comment]
It looks like https://github.com/hashicorp/packer-plugin-amazon/pull/179 (which includes the fix for https://github.com/hashicorp/packer-plugin-amazon/issues/144) was merged yesterday. 

Not trying to hurry anything along, but any guestimates to when we might see this land in a Packer release?

It's been a minor pain point for us as we're trying to provision NixOS 21.11 AMIs using Packer and their configuration of OpenSSH won't accept RSA host keys.

**Comment by JenGoldstrich** [comment]
Hey all, thanks for reporting this @amrragab8080 and to everyone else for jumping in and contributing, and huge thanks to @wedge-jarrad for fixing this issue in https://github.com/hashicorp/packer-plugin-amazon/issues/144.

To answer @programmablemike's question we have released v1.0.7 of the Amazon Packer Plugin, this should resolve this issue https://github.com/hashicorp/packer-plugin-amazon/releases/tag/v1.0.7.  Can y'all let me know if this fixes your issues @amrragab8080 and @programmablemike as it should have.

**Comment by programmablemike** [comment]
Awesome. I can confirm this fixed the issue for us with building NixOS 21.11.

Thanks @JenGoldstrich and the Packer team!

**Comment by nywilken** [comment]
👋 @JenGoldstrich thanks for pinging the thread, and thanks to everyone for testing the fix. We just released [v1.0.8](https://github.com/hashicorp/packer-plugin-amazon/releases/tag/v1.0.8) to address a small regression when setting temporary_key_pair_name but no temporary_key_pair_type. I'm going to close this issue as it has been resolved. 

If you run into any issues with the latest release please drop us a line on the thread and we will gladly reopen. Cheers!
